job_creator.php

```diff
@@ -0,0 +1,273 @@
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        // e.g. ['4', '11']
+        $portions = explode('.', $branch);
+        if (count($portions) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $portions[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            // remove '4' version
+            $installerVersions = array_diff($installerVersions, ['4']);
+            // get the minor portions of the verisons e.g. [9, 10, 11]
+            $minorPortions = array_map(fn($portions) => (int) explode('.', $portions)[1], $installerVersions);
+            sort($minorPortions);
+            return '4.' . $minorPortions[count($minorPortions) - 1] . '.x-dev';
+        }
+    }
+    
+    public function createJob(int $phpIndex, array $opts): array
+    {
+        $installerKey = str_replace('.x-dev', '', $this->installerVersion);
+        $installerKey = $installerKey ?: '4';
+        $phpVersions = INSTALLER_TO_PHP_VERSIONS[$installerKey];
+        $default = [
+            # ensure there's a default value for all possible return keys
+            # this allows us to use `if [[ "${{ matrix.key }}" == "true" ]]; then` in github-actions-ci-cd/ci.yml
+            'installer_version' => $this->installerVersion,
+            'php' => $phpVersions[$phpIndex] ?? $phpVersions[count($phpVersions) - 1],
+            'db' => DB_MYSQL_57,
+            'composer_require_extra' => '',
+            'composer_args' => '',
+            'name_suffix' => '',
+            'phpunit' => false,
+            'phpunit_suite' => 'all',
+            'phplinting' => false,
+            'phpcoverage' => false,
+            'endtoend' => false,
+            'endtoend_suite' => 'root',
+            'endtoend_config' => '',
+            'js' => false,
+        ];
+        return array_merge($default, $opts);
+    }
+    
+    private function parseBoolValue(mixed $value): bool
+    {
+        return ($value === true || $value === 'true');
+    }
+
+    private function createPhpunitJobs(array $matrix, bool $simpleMatrix, string $suite): array
+    {
+        if ($simpleMatrix) {
+            $matrix['include'][] = $this->createJob(0, [
+                'phpunit' => true,
+                'phpunit_suite' => $suite,
+            ]);
+        } else {
+            $matrix['include'][] = $this->createJob(0, [
+                'composer_args' => '--prefer-lowest',
+                'phpunit' => true,
+                'phpunit_suite' => $suite,
+            ]);
+            $matrix['include'][] = $this->createJob(1, [
+                'db' => DB_PGSQL,
+                'phpunit' => true,
+                'phpunit_suite' => $suite,
+            ]);
+            $matrix['include'][] = $this->createJob(3, [
```

Comment:
```suggestion
            $matrix['include'][] = $this->createJob(2, [
                'db' => DB_MYSQL_57_PDO,
                'phpunit' => true,
                'phpunit_suite' => $suite,
            ]);
            $matrix['include'][] = $this->createJob(3, [
```
As per [this comment chain](https://github.com/silverstripe/gha-generate-matrix/pull/1#discussion_r891903314)

Reply:
This job is created in the phpcoverage block, so I've updated code so that this is only add this is not creating a phpcoverage job